application/dash_application/dash_id.py

```diff
@@ -1,9 +1,14 @@
-import uuid
+from inspect import getmodule, stack
 
-### Helps with graph UUIDs
+### Helps with graph IDs
+## Dash requires unique IDs for the entire app. Reduce mental overhead by 
+## generating IDs with module names appended, effectively making IDs
+## appear local to module.
 
 def init_ids(names):
     results = {}
+    callingModule = getmodule(stack()[1][0])
```

Comment:
I don't really grok the stack, but would it not be dangerous to depend on the interpreter stack location? For example, if a refactor wraps `init_ids` in another function (for example a base class for a Dashi app), would this not break? This could be fixed by changing the index in the function, but that still means that some callers might need index `1`, while other would need index `2`.

Reply:
You're right, however from searching online this is the only way to get the name of the calling module in Python, short of passing it in manually which would introduce its own set of problems. This is why I was leaning towards UUIDs when I designed dash_id. Opened this PR because there had been multiple requests for using module names instead

Reply:
Yeah, this is kind of “Thanks, I hate it”, but I think it will work well in most cases. Can you make it a parameter with a default, so we can use a value if necessary and we get the stack-inspected value otherwise?

Reply:
Thanks @apmasell , would you want the parameter to pass the name we want to append to the ID, or the stack location as int? I could see pros and cons to each (string is a lot clearer, int might let us do weird recursive increment logic if we really wanted)

Reply:
Another approach is to have more control over IDs and have Dash crash if there are collisions (which can be tested for). Currently, this does not happen because `app.config.suppress_callback_exceptions = True` is set due to Multipage sadness. However, there is an approach at https://dash.plot.ly/urls (Dynamically Create a Layout for Multi-Page App Validation) that allows for the suppression kludge to be removed.

This would allow us to set our own ID structure (prepend __name__) and any inadvertent collisions would be apparent rather than lead to undefined behaviour.

Reply:
The name that you append. It's more clear if it it's `init_ids(..., "some weird name")` than an arbitrary number.

Reply:
To be clear about removing `app.config.suppress_callback_exceptions = True`. The crash would happen very early in Dash startup, rather than when one of the conflicting callbacks is fired.

Reply:
@apmasell done

@slazicoicr Just tested by cherry-picking this change to the `example` branch. Setting 'moduleName' to the same thing in two different pages causes Dash to crash on calling `flask run`. Serving placeholder layouts to the validator might actually cause the behaviour you're looking to avoid.